### PR TITLE
223 process detail time line

### PR DESCRIPTION
**Pull request by ricardovdheijden**
#223 , Rendering a timeline for the steps in a process.

In case a step is retried or in some way executed twice, both steps get visually combined with a '2' in the timeline:
image

### CODE REVIEW

File: packages/orchestrator-ui-components/src/components/WFOTimeline/WFOTimeline.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    const mapTimelineItemToEuiStep = ({
+        processStepStatus,
+        value,
+        onClick,
+    }: TimelineItem): EuiStepHorizontalProps => {
+        const euiStepStatus =
+            mapProcessStepStatusToEuiStepStatus(processStepStatus);
+
+        return {
+            status: typeof value === 'number' ? undefined : euiStepStatus,
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/components/WFOTimeline/WFOTimeline.tsx:67]: What does this do? It returns undefined if the status is a number but when does this situation occur?  It feels this could be handled in a different place

File: packages/orchestrator-ui-components/src/components/WFOTimeline/mapProcessStepStatusToEuiStepStatus.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+import { EuiStepStatus } from '@elastic/eui/src/components/steps/step_number';
+
+export const mapProcessStepStatusToEuiStepStatus = (
+    processStepStatus: StepStatus,
+): EuiStepStatus => {
+    switch (processStepStatus) {
+        case StepStatus.SUCCESS:
+        case StepStatus.SKIPPED:
+        case StepStatus.COMPLETE:
+            return 'complete';
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/components/WFOTimeline/mapProcessStepStatusToEuiStepStatus.ts:11]: Does Eui expose an enum for the step statusses that you can use for the return value?

File: packages/orchestrator-ui-components/src/pages/processes/timelineUtils.ts [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+            console.log(`Clicked on ${step.name} (${step.stepid})`);
+        },
+    };
+};
+
+const stepsShouldBeMerged = (
+    previousStep: ProcessDetailStep,
+    currentStep: ProcessDetailStep,
+) =>
+    previousStep.status !== StepStatus.COMPLETE &&
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/pages/processes/timelineUtils.ts:27]: I would like to see a test for this logic. I think StepStatus.SUCCES should also be checked for the edge case you have 2 steps with the same name in succession.

Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/pages/processes/timelineUtils.ts:27]: Good one, an probably SKIPPED as well

File: packages/orchestrator-ui-components/src/components/WFOTimeline/WFOTimeline.tsx (hunk 1 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/components/WFOTimeline/WFOTimeline.tsx]: I made the results to be returned a bit more explicit (3 cases)

File: packages/orchestrator-ui-components/src/pages/processes/timelineUtils.ts (hunk 3 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/pages/processes/timelineUtils.ts]: Introduced a helper function to determine if a status is considered final

**Review by DutchBen — APPROVED** [review]

File: packages/orchestrator-ui-components/src/components/WFOTimeline/styles.ts [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        display: 'flex',
+        flexDirection: 'row',
+        alignItems: 'center',
+    });
+
+    return {
+        timelinePanelStyle,
+        stepStyle,
+        getStepLineStyle,
+        stepOuterCircleStyle: getStepOuterCircleStyle,
```
Comment (DutchBen) [inline packages/orchestrator-ui-components/src/components/WFOTimeline/styles.ts:151]: why the rename?

**Review by DutchBen — APPROVED** [review]

File: packages/orchestrator-ui-components/src/components/WFOTimeline/styles.ts (hunk 4 above)
Comment (ricardovdheijden) [reply packages/orchestrator-ui-components/src/components/WFOTimeline/styles.ts]: It used to be plain style before , but now its a function. Refactor in the IDE left the previous name to maintain the interface for the consuming components. -- Not needed, fixed it now
